cli/src/commands/version.rs

```diff
@@ -22,14 +22,66 @@ use crate::ui::Ui;
 
 /// Display version information
 #[derive(clap::Args, Clone, Debug)]
-pub(crate) struct VersionArgs {}
+pub(crate) struct VersionArgs {
+    /// Display only the version number and nothing else.
+    #[arg(long)]
+    pub(crate) numeric_only: bool,
+
+    /// Display build information.
+    #[arg(long, conflicts_with = "numeric_only")]
+    pub(crate) build_info: bool,
+}
 
 #[instrument(skip_all)]
 pub(crate) fn cmd_version(
     ui: &mut Ui,
     command: &CommandHelper,
-    _args: &VersionArgs,
+    args: &VersionArgs,
 ) -> Result<(), CommandError> {
-    write!(ui.stdout(), "{}", command.app().render_version())?;
+    let base_version = command.app().get_version().unwrap();
+    let (version, git_commit) = if let Some(git_commit) = option_env!("JJ_GIT_COMMIT") {
+        // in a release build, don't include the commit hash in the
+        // `--numeric-version` or normal output. GH-3629
+        if option_env!("JJ_RELEASE_BUILD").is_some() {
+            (String::from(base_version), git_commit)
+        } else {
+            let short_commit = &git_commit[..12];
+            (format!("{}-{}", base_version, short_commit), git_commit)
+        }
+    } else {
+        (String::from(base_version), "unknown")
+    };
+
+    if args.numeric_only {
+        writeln!(ui.stdout(), "{}", version)?;
+        return Ok(());
+    }
+
+    write!(
+        ui.stdout(),
+        "\
+Jujutsu version control system; jj {}
```

Comment:
I would prefer removing all the legalese (copyright, license, warranty). Only the GNU project seems to like doing things that way, and I'd prefer just not doing them until a lawyer tells us to.

Reply:
I've shuffled it around a little, but I sort of like the "heft" to it, as well &mdash; the copyright line gives some notion of the author, etc. It's vibes, in a way.